src/dune_engine/fs_memo.ml

```diff
@@ -280,15 +280,20 @@ let dir_contents ?(force_update = false) path =
    because the result's type depends on [f]. There are only two call sites of
    [with_lexbuf_from_file], so perhaps we could just replace this more general
    function with two simpler ones that can be cached independently. *)
-let with_lexbuf_from_file path ~f =
+let make_reader path ~f =
```

Comment:
The comment should be moved.

Also, I'd give this function a more explicit name like `tracking_file_digest`, because that's what it's doing.